hugr-core/src/extension/resolution/types_mut.rs

```diff
@@ -0,0 +1,212 @@
+//! Resolve weak links inside `CustomType`s in an optype's signature, while
+//! collecting all used extensions.
+//!
+//! For a non-mutating option see [`super::collect_op_types_extensions`].
```

Comment:
I reckon the pub function here should be `OpType::used_extensions`.

Reply:
It's a private module. (Otherwise the docs check would complain)

Reply:
Ah ok, yes I missed that. Ok I'll hold some fire there then ;)